crates/forge_loader/src/manifest.rs

```diff
@@ -59,23 +81,159 @@ enum Interval {
     Week,
 }
 
+// Maps to Scheduled Trigger under Common Modules
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
 struct ScheduledTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
     interval: Interval,
 }
 
+// Maps to Web Trigger under Common Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct RawTrigger<'a> {
+    key: &'a str,
+    function: &'a str,
+}
+
+// maps to Trigger under Common Modules
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
-struct WorkflowPostFunction<'a> {
+struct EventTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
+    #[serde(borrow)]
+    events: Vec<&'a str>,
 }
 
+// Compass Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CompassAdminPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct ComponentPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct DataProvider<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    callback: JustFunc<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CompassGlobalPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct TeamPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+// Confluence Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct ContentAction<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct ContentByLineItem<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    #[serde(borrow)]
+    dynamic_properties: JustFunc<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct MacroMod<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    config: JustFunc<'a>,
+    export: JustFunc<'a>,
+}
+
+// Jira Modules
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct JiraAdminPage<'a> {
+    key: &'a str,
+    function: &'a str,
+    title: &'a str,
+    resolver: Resolver<'a>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CustomField<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    value: Option<&'a str>,
+    search_suggestions: Option<&'a str>,
+    edit: Option<&'a str>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CustomFieldType<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    value: Option<&'a str>,
+    edit: Option<&'a str>,
+    context_config: Option<&'a str>,
```

Comment:
Also, since `context_config` is camelCase in the manifest, you'll need to either add `#[serde(rename = "contextConfig")]` above it or add `#[serde(rename_all = "camelCase")]` at the top of the struct right below the derive.

Reply:
It might also be nice to leave a comment with a link to [this page](https://developer.atlassian.com/platform/forge/manifest-reference/modules/jira-custom-field-type/#search-suggestions) on top of the struct, explaining that `searchSuggestions` isn't user invokable, which is why we don't deserialize it. Rustdoc renders markdown, so you can use markdown embed linking in your comment if you want.